File: from-3401-to-3500/3456-find-special-substring-of-length-k/solution.go
func hasSpecialSubstring(s string, k int) bool {
    if k == 1 && (len(s) == 1 || s[0] != s[1]) {
        return true
    }
    prev := s[0]
    same := 1
    for i := 1; i < len(s); i ++ {
        curr := s[i]
        if curr == prev {
            same += 1
        } else {
            if same == k {
                return true
            }
            same = 1
        }
        prev = curr
    }
    if same == k {
        return true
    }
    return false
}

File: from-3401-to-3500/3456-find-special-substring-of-length-k/solution.py
class Solution:
    def hasSpecialSubstring(self, s: str, k: int) -> bool:
        if k == 1 and (len(s) == 1 or s[0] != s[1]):
            return True
        prev = s[0]
        same = 1
        for curr in s[1:]:
            if curr == prev:
                same += 1
            else:
                if same == k:
                    return True
                same = 1
            prev = curr
        if same == k:
            return True
        return False